lib/measures/tbd/measure.rb

```diff
@@ -130,6 +138,76 @@ def run(model, runner, user_arguments)
       end
     end
 
+    if gen_kiva
+      # Although one may choose to auto-generate Kiva settings and objects,
+      # there must be at least one valid foundation-facing floor in the model.
+      kiva = false
+      surfaces.values.each do |surface|
+        next if kiva
+        kiva = true if surface.has_key?(:kiva)
+      end
+
+      if kiva
+        arg = "TotalExposedPerimeter"
+        foundation_kiva_settings = model.getFoundationKivaSettings
+        foundation_kiva_settings.setName("TBD-generated Kiva settings template")
+
+        # Generic 1" XPS insulation.
+        xps_25mm = OpenStudio::Model::StandardOpaqueMaterial.new(model)
+        xps_25mm.setName("XPS_25mm")
+        xps_25mm.setRoughness("Rough")
+        xps_25mm.setThickness(0.0254)
+        xps_25mm.setConductivity(0.029)
+        xps_25mm.setDensity(28)
+        xps_25mm.setSpecificHeat(1450)
+        xps_25mm.setThermalAbsorptance(0.9)
+        xps_25mm.setSolarAbsorptance(0.7)
+
+        # Typical circa-1980 slab-on-grade (perimeter) insulation setup.
+        kiva_slab = OpenStudio::Model::FoundationKiva.new(model)
+        kiva_slab.setName("Kiva slab")
+        kiva_slab.setInteriorHorizontalInsulationMaterial(xps_25mm)
+        kiva_slab.setInteriorHorizontalInsulationWidth(0.6)
+
+        # Basement wall setup (full-height insulation as construction layer)
+        kiva_basement = OpenStudio::Model::FoundationKiva.new(model)
+        kiva_basement.setName("Kiva basement")
```

Comment:
If there are foundation walls linked to the foundation slab, then TBD presumes a basement setup. If not, a slab-on-grade setup.